sync/result_pool.go

```diff
@@ -0,0 +1,127 @@
+// Package sync provides ways to execute multiple tasks concurrently
+// while synchronously managing them at the same time.
+package sync
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/getsentry/sentry-go"
+	"github.com/sourcegraph/conc/panics"
+	"github.com/sourcegraph/conc/pool"
+)
+
+type ResultPool[T any] struct {
+	pool *pool.ResultContextPool[T]
+	span *sentry.Span
+}
+
+type ResultPoolOption func(*resultPoolOptions)
+
+type resultPoolOptions struct {
+	pl                 poolOptions
+	collectErroredRlts bool
+}
+
+func WithRltMaxGoroutines(max int) ResultPoolOption {
+	return func(opts *resultPoolOptions) {
+		opts.pl.max = max
+	}
+}
+
+func WithRltRequest(req *http.Request) ResultPoolOption {
+	return func(opts *resultPoolOptions) {
+		opts.pl.req = req
+	}
+}
+
+func WithRltCancelOnFirstErr() ResultPoolOption {
+	return func(opts *resultPoolOptions) {
+		opts.pl.cancelOnFirstErr = true
+	}
+}
+
+func WithCollectErroredRlts() ResultPoolOption {
+	return func(opts *resultPoolOptions) {
+		opts.collectErroredRlts = true
+	}
+}
+
+func NewResultPool[T any](ctx context.Context, opts ...ResultPoolOption) ResultPool[T] {
+
+	// set default options
+	plOpts := &resultPoolOptions{
+		pl: poolOptions{
+			req: &http.Request{
+				Method: "",
+				URL:    &url.URL{},
+			},
+			max:              0,
+			cancelOnFirstErr: false,
+		},
+		collectErroredRlts: false,
+	}
+
+	for _, opt := range opts {
+		opt(plOpts)
+	}
+
+	// set setry transaction or span
+	span := setSpan(ctx, plOpts.pl.req)
+	if span != nil {
+		ctx = span.Context()
+	}
+
+	var pl *pool.ResultContextPool[T]
+	if plOpts.pl.cancelOnFirstErr {
+		pl = pool.NewWithResults[T]().WithContext(ctx).WithFirstError().WithCancelOnError()
+	} else {
+		pl = pool.NewWithResults[T]().WithErrors().WithContext(ctx)
+	}
+	if plOpts.collectErroredRlts {
+		pl = pl.WithCollectErrored()
+	}
+	if plOpts.pl.max > 0 {
+		pl.WithMaxGoroutines(plOpts.pl.max)
+	}
+
+	return ResultPool[T]{
+		pool: pl,
+		span: nil,
```

Comment:
should be `span`

Reply:
Really thanks for reviewing!
Fixed both as you pointed out.